.grit/patterns/js/iots_to_zod.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Transform `io-ts` to `zod schema`
+---
+
+Transform `io-ts` schemas to `zod` schema
+
+tags: #migration, #js, #zod, #io-ts
+
+```grit
+engine marzano(0.1)
+language js
+
+or {
+    `import * as $alias from "io-ts"` => `import z from 'zod'`,
+    `$alias.$param` => `z.$param()`,
```

Comment:
This will match *any* `$x.$y`, since it is *not* constrained by the first part. Ex. it would turn `console.log` into `z.log()`.

You need to first find the $alias then use it in the rest of the `$program`. Like so:

```
engine marzano(0.1)
language js

`import * as $alias from "io-ts"` => `import z from 'zod'` where {
   $program <: contains bubble($alias) or {
     `$alias.$param` => `z.$param()`,
      `$alias.type($val)` => `z.object($val)`,
      `$alias.$param($val)` => `z.$param($val)`,
   }
}
```

Reply:
Make a lot of sense! I added a simple object and function to test to ensure they will not be changed. 
Along with the recent changes, I've added initial support for `t.partial`.